lib/masamune/filesystem.rb

```diff
@@ -157,6 +157,22 @@ def touch!(*files)
       end
     end
 
+    def size(file)
+      case type(file)
+      when :hdfs
+        hadoop_fs('-test', '-z', file)
+      when :s3
+        result = 0
+        s3cmd('ls', s3b(file), safe: true) do |line|
+          _date, _time, size, _name = line.split(/\s+/)
+          result = size
+        end
+        result.to_i
+      when :local
+        File.size(file)
+      end
+    end
```

Comment:
`Filesystem#stat` can get the file size from various remote file systems- it also has some logic to handle remote directories.  I think this code could be reused as follows:
```
def size(file_or_dir)
  stat(file_or_dir).try(:size)
end
```